test/uk/gov/hmrc/cdsimportsddsfrontend/services/CustomsDeclarationsServiceSpec.scala

```diff
@@ -49,21 +50,26 @@ class CustomsDeclarationsServiceSpec extends WordSpec
     "Post a raw XML declaration to the Declaration API" in new Scenario() {
       val decApiResponse = CustomsDeclarationsResponse(200, Some("conversation id"))
       when[Future[CustomsDeclarationsResponse]](mockHttp.POSTString(any(),any(),any())(any(), any(), any())).thenReturn(Future.successful(decApiResponse))
-      val response: DeclarationServiceResponse = await(customsDeclarationsService.submit(testEori, <DeclaringMyStuff/>))
+      val response: DeclarationServiceResponse = await(customsDeclarationsService.submit(testEori, <DeclaringMyStuff/>, None))
 
       response.conversationId mustBe decApiResponse.conversationId
       response.status mustBe decApiResponse.status
       response.xml mustBe "&lt;DeclaringMyStuff/&gt;"
     }
 
-    "maps view model to declaration and post to the Declaration API" in new Scenario() {
+    "maps view model to declaration, post to the Declaration API and save the submitted declaration" in new Scenario() {
```

Comment:
Should we have another test that checks that the backend is not called when the call to the DEC API fails?

Reply:
sure, will add one